File: Python/Program/Garage.py
from Vehicle import Vehicle
from ParkingLot import ParkingLot

class Garage(ParkingLot):
    def __init__(self, nama_garasi, luas_garasi, kapasitas, jumlah_kendaraan_saat_ini):
        super().__init__(kapasitas, 0)
        self.nama_garasi = nama_garasi
        self.luas_garasi = luas_garasi
        self.daftar_kendaraan = []

    def get_nama_garasi(self):
        return self.nama_garasi

    def set_nama_garasi(self, nama_garasi):
        self.nama_garasi = nama_garasi

    def get_luas_garasi(self):
        return self.luas_garasi

    def set_luas_garasi(self, luas_garasi):
        self.luas_garasi = luas_garasi

    def get_daftar_kendaraan(self):
        return self.daftar_kendaraan

    def tambah_kendaraan(self, vehicle):
        if len(self.daftar_kendaraan) < self.get_kapasitas():
            self.daftar_kendaraan.append(vehicle)
            self.set_jumlah_kendaraan_saat_ini(len(self.daftar_kendaraan))
            print(f"Kendaraan {vehicle.get_plat_nomor()} berhasil ditambahkan ke dalam garasi.")
        else:
            print("Kapasitas garasi tidak mencukupi. Penambahan kendaraan ditolak.")
            
    def hapus_kendaraan(self, vehicle):
        self.daftar_kendaraan.remove(vehicle)
        self.set_jumlah_kendaraan_saat_ini(len(self.daftar_kendaraan))
        print(f"Kendaraan {vehicle.get_plat_nomor()} berhasil dihapus dari garasi.")

File: CPP/Program/Garage.cpp
// Menggunakan pragma once agar header file hanya di-include satu kali
#pragma once
// Memasukan library yang digunakan
#include <iostream>
#include <string>
#include <vector>
#include "ParkingLot.cpp" // Assuming ParkingLot class is defined in ParkingLot.h
#include "Vehicle.cpp"    // Assuming Vehicle class is defined in Vehicle.h

// Deklarasi namespace
using namespace std;

class Garage : public ParkingLot {
private:
    string nama_garasi;
    double luas_garasi;
    vector<Vehicle*> daftar_kendaraan;
public:
    Garage(string nama_garasi, double luas_garasi, int kapasitas, int jumlah_kendaraan_saat_ini) : ParkingLot(kapasitas, 0), nama_garasi(nama_garasi), luas_garasi(luas_garasi) {}

    string getNamaGarasi() {
        return nama_garasi;
    }

    void setNamaGarasi(string nama_garasi) {
        this->nama_garasi = nama_garasi;
    }

    double getLuasGarasi() const {
        return luas_garasi;
    }

    void setLuasGarasi(double luas_garasi) {
        this->luas_garasi = luas_garasi;
    }

    vector<Vehicle*>& getDaftarKendaraan() {
        return daftar_kendaraan;
    }

    void tambahKendaraan(Vehicle* vehicle) {
        if (daftar_kendaraan.size() < getKapasitas()) {
            daftar_kendaraan.push_back(vehicle); // Menggunakan pointer vehicle
            setJumlahKendaraanSaatIni(daftar_kendaraan.size());
            std::cout << "Kendaraan " << vehicle->getPlatNomor() << " berhasil ditambahkan ke dalam garasi." << std::endl;
        }
         else {
            std::cout << "Kapasitas garasi tidak mencukupi. Penambahan kendaraan ditolak." << std::endl;
        }
    }
    
    // Destructor
    ~Garage()
    {

    }
};